### ISSUE TITLE
False positive for `print_literal` with `file!()` macro

### ISSUE THREAD

**Issue by Urhengulas** [rust-lang/rust-clippy#9810]
### Summary

The code snipped below triggers a false `clippy::print_literal` warning with a suggestion that changes the program's output.

### Lint Name

clippy::print_literal

### Reproducer

I tried this code ([playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=c22a2927482cb178d7f1706155d34069)):

```rust
fn main() {
    println!("{}", file!());
}
```

I saw this happen:

```
warning: literal with an empty format string
 --> src/main.rs:2:20
  |
2 |     println!("{}", file!());
  |                    ^^^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#print_literal
  = note: `#[warn(clippy::print_literal)]` on by default
help: try this
  |
2 -     println!("{}", file!());
2 +     println!("ile!(");
  |
```

I expected to see this happen:

no lint

### Version

```text
rustc 1.67.0-nightly (1286ee23e 2022-11-05)
binary: rustc
commit-hash: 1286ee23e4e2dec8c1696d3d76c6b26d97bbcf82
commit-date: 2022-11-05
host: x86_64-unknown-linux-gnu
release: 1.67.0-nightly
LLVM version: 15.0.4
```

**Comment by y21** [comment]
This issue seems to be fixed, the provided code doesn't seem to trigger the lint anymore.
